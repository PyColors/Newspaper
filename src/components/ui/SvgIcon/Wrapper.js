import styled from 'styled-components';
import theme from '../../../styles/theme';

const { fontSize, colors } = theme;

const Wrapper = styled.svg`
  font-size: ${fontSize.xlarge};
  color: ${colors.white};
  width: ${fontSize.xlarge};
  height: ${fontSize.xlarge};
`;

export default Wrapper;
