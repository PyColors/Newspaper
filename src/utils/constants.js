import theme from '../styles/theme';
const { colors } = theme;

export const ICONS = {
  USER:
    'M18 22.082v-1.649c2.203-1.241 4-4.337 4-7.432 0-4.971 0-9-6-9s-6 4.029-6 9c0 3.096 1.797 6.191 4 7.432v1.649c-6.784 0.555-12 3.888-12 7.918h28c0-4.030-5.216-7.364-12-7.918z',
  GOOGLE:
    'M522.2 438.8v175.6h290.4c-11.8 75.4-87.8 220.8-290.4 220.8-174.8 0-317.4-144.8-317.4-323.2s142.6-323.2 317.4-323.2c99.4 0 166 42.4 204 79l139-133.8c-89.2-83.6-204.8-134-343-134-283 0-512 229-512 512s229 512 512 512c295.4 0 491.6-207.8 491.6-500.2 0-33.6-3.6-59.2-8-84.8l-483.6-0.2z',
  FACEBOOK:
    'M608 192h160v-192h-160c-123.514 0-224 100.486-224 224v96h-128v192h128v512h192v-512h160l32-192h-192v-96c0-17.346 14.654-32 32-32z',
  TWITTER:
    'M1024 226.4c-37.6 16.8-78.2 28-120.6 33 43.4-26 76.6-67.2 92.4-116.2-40.6 24-85.6 41.6-133.4 51-38.4-40.8-93-66.2-153.4-66.2-116 0-210 94-210 210 0 16.4 1.8 32.4 5.4 47.8-174.6-8.8-329.4-92.4-433-219.6-18 31-28.4 67.2-28.4 105.6 0 72.8 37 137.2 93.4 174.8-34.4-1-66.8-10.6-95.2-26.2 0 0.8 0 1.8 0 2.6 0 101.8 72.4 186.8 168.6 206-17.6 4.8-36.2 7.4-55.4 7.4-13.6 0-26.6-1.4-39.6-3.8 26.8 83.4 104.4 144.2 196.2 146-72 56.4-162.4 90-261 90-17 0-33.6-1-50.2-3 93.2 59.8 203.6 94.4 322.2 94.4 386.4 0 597.8-320.2 597.8-597.8 0-9.2-0.2-18.2-0.6-27.2 41-29.4 76.6-66.4 104.8-108.6z',
  GITHUB:
    'M409.132,114.573c-19.608-33.596-46.205-60.194-79.798-79.8C295.736,15.166,259.057,5.365,219.271,5.365   c-39.781,0-76.472,9.804-110.063,29.408c-33.596,19.605-60.192,46.204-79.8,79.8C9.803,148.168,0,184.854,0,224.63   c0,47.78,13.94,90.745,41.827,128.906c27.884,38.164,63.906,64.572,108.063,79.227c5.14,0.954,8.945,0.283,11.419-1.996   c2.475-2.282,3.711-5.14,3.711-8.562c0-0.571-0.049-5.708-0.144-15.417c-0.098-9.709-0.144-18.179-0.144-25.406l-6.567,1.136   c-4.187,0.767-9.469,1.092-15.846,1c-6.374-0.089-12.991-0.757-19.842-1.999c-6.854-1.231-13.229-4.086-19.13-8.559   c-5.898-4.473-10.085-10.328-12.56-17.556l-2.855-6.57c-1.903-4.374-4.899-9.233-8.992-14.559   c-4.093-5.331-8.232-8.945-12.419-10.848l-1.999-1.431c-1.332-0.951-2.568-2.098-3.711-3.429c-1.142-1.331-1.997-2.663-2.568-3.997   c-0.572-1.335-0.098-2.43,1.427-3.289c1.525-0.859,4.281-1.276,8.28-1.276l5.708,0.853c3.807,0.763,8.516,3.042,14.133,6.851   c5.614,3.806,10.229,8.754,13.846,14.842c4.38,7.806,9.657,13.754,15.846,17.847c6.184,4.093,12.419,6.136,18.699,6.136   c6.28,0,11.704-0.476,16.274-1.423c4.565-0.952,8.848-2.383,12.847-4.285c1.713-12.758,6.377-22.559,13.988-29.41   c-10.848-1.14-20.601-2.857-29.264-5.14c-8.658-2.286-17.605-5.996-26.835-11.14c-9.235-5.137-16.896-11.516-22.985-19.126   c-6.09-7.614-11.088-17.61-14.987-29.979c-3.901-12.374-5.852-26.648-5.852-42.826c0-23.035,7.52-42.637,22.557-58.817   c-7.044-17.318-6.379-36.732,1.997-58.24c5.52-1.715,13.706-0.428,24.554,3.853c10.85,4.283,18.794,7.952,23.84,10.994   c5.046,3.041,9.089,5.618,12.135,7.708c17.705-4.947,35.976-7.421,54.818-7.421s37.117,2.474,54.823,7.421l10.849-6.849   c7.419-4.57,16.18-8.758,26.262-12.565c10.088-3.805,17.802-4.853,23.134-3.138c8.562,21.509,9.325,40.922,2.279,58.24   c15.036,16.18,22.559,35.787,22.559,58.817c0,16.178-1.958,30.497-5.853,42.966c-3.9,12.471-8.941,22.457-15.125,29.979   c-6.191,7.521-13.901,13.85-23.131,18.986c-9.232,5.14-18.182,8.85-26.84,11.136c-8.662,2.286-18.415,4.004-29.263,5.146   c9.894,8.562,14.842,22.077,14.842,40.539v60.237c0,3.422,1.19,6.279,3.572,8.562c2.379,2.279,6.136,2.95,11.276,1.995   c44.163-14.653,80.185-41.062,108.068-79.226c27.88-38.161,41.825-81.126,41.825-128.906   C438.536,184.851,428.728,148.168,409.132,114.573z',
  ARROW_RIGHT:
    'm40.4,121.3c-0.8,0.8-1.8,1.2-2.9,1.2s-2.1-0.4-2.9-1.2c-1.6-1.6-1.6-4.2 0-5.8l51-51-51-51c-1.6-1.6-1.6-4.2 0-5.8 1.6-1.6 4.2-1.6 5.8,0l53.9,53.9c1.6,1.6 1.6,4.2 0,5.8l-53.9,53.9z'
};

export const COLOR_ICONS = {
  BLACK: `${colors.black}`,
  WHITE: `${colors.white}`
};
