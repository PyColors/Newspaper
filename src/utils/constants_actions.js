/* Actions */

/* Common */
export const CLEAR_STORE = 'CLEAR_STORE';

/* Home Page */
export const FETCH_DATA = 'FETCH_DATA';
export const RECEIVE_DATA = 'RECEIVE_DATA';
export const FAILED_RECIEVE_DATA = 'FAILED_RECIEVE_DATA';
