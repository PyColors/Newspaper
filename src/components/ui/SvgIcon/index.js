import React from 'react';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';

const SvgIcon = props => {
  return (
    <Wrapper viewBox="0 0 1024 1024">
      <g fill={props.color}>
        <path d={props.icon} />
      </g>
    </Wrapper>
  );
};

SvgIcon.propTypes = {
  color: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
};

export default SvgIcon;
