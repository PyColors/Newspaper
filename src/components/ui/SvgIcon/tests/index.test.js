import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SvgIcon from '../index';
import { COLOR_ICONS, ICONS } from '../../../../utils/constants';

const icon = ICONS.FACEBOOK;
const color = COLOR_ICONS.BLACK;
const viewBox = '0 0 1024 1024';

const renderComponent = (props = {}) =>
  shallow(<SvgIcon color={color} icon={icon} {...props} />);

describe('<SvgIcon />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(<SvgIcon color={COLOR_ICONS.BLACK} icon={ICONS.FACEBOOK} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a viewBox attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('viewBox')).toEqual(viewBox);
  });

  it('should not render texts', () => {
    const children = 'Text';
    const renderedComponent = shallow(<SvgIcon>{children}</SvgIcon>);
    expect(renderedComponent.contains(children)).toBe(false);
  });

  it('should not have a className attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('className')).toBeUndefined();
  });
});
