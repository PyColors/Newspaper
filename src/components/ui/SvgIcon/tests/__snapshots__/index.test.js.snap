// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon /> should match the snapshot 1`] = `
<svg
  className="sc-bdVaJa fkiftl"
  viewBox="0 0 1024 1024"
>
  <g
    fill="#222"
  >
    <path
      d="M608 192h160v-192h-160c-123.514 0-224 100.486-224 224v96h-128v192h128v512h192v-512h160l32-192h-192v-96c0-17.346 14.654-32 32-32z"
    />
  </g>
</svg>
`;
